packages/protocol/engine/tangle/booker/booker.go

```diff
@@ -450,61 +470,102 @@ func (b *Booker) setupEvents() {
 	}, b.workers.CreatePool("Booker", 2))
 }
 
+func (b *Booker) OrphanAttachment(block *Block) {
+	if tx, isTx := block.Transaction(); isTx {
+		attachmentBlock, attachmentOrphaned, lastAttachmentOrphaned := b.attachments.AttachmentOrphaned(tx.ID(), block)
+
+		if attachmentOrphaned {
+			b.Events.AttachmentOrphaned.Trigger(attachmentBlock)
+		}
+
+		if lastAttachmentOrphaned {
+			// TODO: attach this event somewhere to the engine
+			b.Events.Error.Trigger(errors.Errorf("transaction %s orphaned", tx.ID()))
+			b.Ledger.PruneTransaction(tx.ID(), true)
+		}
+	}
+}
+
 // region FORK LOGIC ///////////////////////////////////////////////////////////////////////////////////////////////////
 
 // PropagateForkedConflict propagates the forked ConflictID to the future cone of the attachments of the given Transaction.
 func (b *Booker) PropagateForkedConflict(transactionID, addedConflictID utxo.TransactionID, removedConflictIDs utxo.TransactionIDs) (err error) {
-	for blockWalker := walker.New[*Block]().PushAll(b.attachments.Get(transactionID)...); blockWalker.HasNext(); {
-		block := blockWalker.Next()
+	blockWalker := walker.New[*Block]()
 
-		updated, propagateFurther, forkErr := b.propagateForkedConflict(block, addedConflictID, removedConflictIDs)
-		if forkErr != nil {
-			blockWalker.StopWalk()
-			return errors.Wrapf(forkErr, "failed to propagate forked ConflictID %s to future cone of %s", addedConflictID, block.ID())
-		}
-		if !updated {
-			continue
-		}
+	for it := b.GetAllAttachments(transactionID).Iterator(); it.HasNext(); {
+		attachment := it.Next()
+		blockWalker.Push(attachment)
 
-		b.Events.BlockConflictAdded.Trigger(&BlockConflictAddedEvent{
-			Block:             block,
-			ConflictID:        addedConflictID,
-			ParentConflictIDs: removedConflictIDs,
-		})
+		blockWalker.PushAll(b.blocksFromBlockDAGBlocks(attachment.WeakChildren())...)
+		blockWalker.PushAll(b.blocksFromBlockDAGBlocks(attachment.LikedInsteadChildren())...)
+	}
 
+	for blockWalker.HasNext() {
+		block := blockWalker.Next()
+		propagateFurther, propagationErr := b.propagateToBlock(block, addedConflictID, removedConflictIDs)
+		if propagationErr != nil {
+			blockWalker.StopWalk()
+			return errors.Wrapf(propagationErr, "failed to propagate forked ConflictID %s to future cone of %s", addedConflictID, block.ID())
+		}
 		if propagateFurther {
-			blockWalker.PushAll(b.strongChildren(block)...)
+			blockWalker.PushAll(b.blocksFromBlockDAGBlocks(block.StrongChildren())...)
 		}
 	}
+
 	return nil
 }
 
-func (b *Booker) propagateForkedConflict(block *Block, addedConflictID utxo.TransactionID, removedConflictIDs utxo.TransactionIDs) (propagated, propagateFurther bool, err error) {
+func (b *Booker) propagateToBlock(block *Block, addedConflictID utxo.TransactionID, removedConflictIDs utxo.TransactionIDs) (propagateFurther bool, err error) {
 	b.bookingMutex.Lock(block.ID())
```

Comment:
Add a comment to explain the mutual exclusion between the `book` -> virtualvoting flow and this `propagateToBlock` -> virtualvoting `processForkedBlock`.
We have to be careful when uncommenting the propagation through markers: we might need something similar there as well to prevent race conditions.